### ISSUE TITLE
BMC Update: Side Switch and Code integration

### ISSUE THREAD

**Issue by anoo1** [openbmc/openbmc#2114]
During BMC update, select the BMC or alternate chip to write the new image, delete any present image beforehand as there's only space for one on each chip.

Pull together all the changes to the BMC filesystem, verify, and add any missing pieces to the end-to-end flow.

**Comment by anoo1** [comment]
- [ ] Change the bootargs u-boot env to use variables so that they can be set by user space during code update. Ex, change from:
```
ubi.mtd=bmc,0,0,0 ro ubi.block=0,1 rootfstype=squashfs root=/dev/ubiblock0_1
```
to
```
bmcmtd=bmc,0,0,0
bmcblock=0,1
bmcroot=/dev/ubiblock0_1
ubi.mtd=$bmcmtd ro ubi.block=$bmcblock root=$bmcroot
```

**Comment by rfrandse** [comment]
<https://gerrit.openbmc-project.xyz/6121> obmc-flash-bmc: Remove existing image before updating
Resolves  #2114 BMC Update: Side Switch and Code integration
